django_app/redbox_app/redbox_core/models.py

```diff
@@ -42,16 +43,63 @@ class ProcessingStatusEnum(models.TextChoices):
     complete = "complete"
 
 
-# TO DO: Based on /redbox/models/file.py, but not complete
-class File(models.Model):
-    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
-    path = models.TextField(help_text="location of file")
-    name = models.TextField()
-    processing_status = models.CharField(choices=ProcessingStatusEnum.choices, default=ProcessingStatusEnum.uploaded)
+class FileTypeEnum(models.TextChoices):
+    eml = ".eml"
+    html = ".html"
+    json = ".json"
+    md = ".md"
+    msg = ".msg"
+    rst = ".rst"
+    rtf = ".rtf"
+    txt = ".txt"
+    xml = ".xml"
+    csv = ".csv"
+    doc = ".doc"
+    docx = ".docx"
+    epub = ".epub"
+    odt = ".odt"
+    pdf = ".pdf"
+    ppt = ".ppt"
+    pptx = ".pptx"
+    tsv = ".tsv"
+    xlsx = ".xlsx"
+    htm = ".htm"
+
+
+class File(UUIDPrimaryKeyBase, TimeStampedModel):
+    processing_status = models.CharField(choices=ProcessingStatusEnum.choices, null=False, blank=False)
+    original_file = models.FileField(storage=settings.BUCKET_NAME)
+    user = models.ForeignKey(User, on_delete=models.CASCADE)
+
+    @property
+    def file_type(self):
+        name = self.original_file.name
+        return name.split(".")[-1]
+
+    @property
+    def file_url(self):
+        return self.original_file.url
+
+    @property
+    def name(self):
+        return self.original_file.name
 
     def get_processing_text(self) -> str:
         processing_status_list = list(ProcessingStatusEnum)
         stage = processing_status_list.index(self.processing_status)
         if stage == len(processing_status_list) - 1:
             return self.processing_status
         return f"{stage + 1}/{len(processing_status_list) - 1} {self.processing_status}"
+
+
+class ChatHistory(UUIDPrimaryKeyBase, TimeStampedModel):
+    name = models.TextField(max_length=1024, null=False, blank=False)
+    users = models.ManyToManyField(User)
+    files_received = models.ForeignKey(File, on_delete=models.CASCADE, related_name="files_received")
+    files_retrieved = models.ForeignKey(File, on_delete=models.CASCADE, related_name="files_retrieved")
+
+
+class ChatMessage(UUIDPrimaryKeyBase, TimeStampedModel):
+    chat_history = models.ForeignKey(ChatHistory, on_delete=models.CASCADE)
+    text = models.TextField(max_length=32768, null=False, blank=False)
+    role = models.TextField(max_length=1024, null=False, blank=False)
```

Comment:
should this be restricted to some enum?

Reply:
What is `role` in this context?

Reply:
I think this is https://github.com/i-dot-ai/redbox-copilot/blob/4dc8ebf96b633d2e0c2ea2a4f0387dc9c39361d9/core_api/src/routes/chat.py#L39

Reply:
It's something expected by various ChatParsers